tools/core-build-tasks/src/tasks/generateContentsJson.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Mojang AB.  All rights reserved.
+
+import { FileSystem } from "@rushstack/node-core-library";
+import path from "path";
+
+/**
+ * Parameters for the generate contents json task.
+ */
+export type GenerateContentsJsonParameters = {
+  /**
+   * The path to the target directory to generate the contents.json file for.
+   */
+  targetPath: string;
+  /**
+   * The path to the output file to write the contents.json file to.
+   */
+  outputFile: string;
+  /**
+   * If true, the task will not throw an error if the target directory does not exist.
+   */
+  ignoreTargetFolderExists?: boolean;
+};
+
+interface Entry {
+  path: string;
+}
+
+interface ContentsData {
+  content: Entry[];
+}
+
+/**
+ * At a given target, runs build script to generate a contents.json for a behavior_pack or resource_pack.
+ */
+export function generateContentsJsonTask(
```

Comment:
Can you add tests for the logic in this task?

Reply:
Still in progress ⌛

Reply:
Done